lib/game.rb

```diff
@@ -46,4 +46,70 @@ def end_game
       puts "----- Tie game -----"
     end
   end
-end
\ No newline at end of file
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  def start
+    @board.welcome_message
+
+    puts "Enter p to play. Enter q to quit.\n"
+    user_input = gets.chomp
+
+    # require 'pry';binding.pry
+    if (user_input.is_a?(Integer) == false)
+      while (user_input.upcase != "Q") && (user_input.upcase != "P")
+        puts "Invalid input.\nEnter p to play. Enter q to quit."
+        user_input = gets.chomp 
+      end
+
+      if user_input.upcase == "Q" 
+        return puts "Thank you for checking out CONNECT FOUR!"
+      end
+
+    else  user_input.upcase == "P"
```

Comment:
If the user were to enter an invalid character followed by a 'P', this else branch will not run as the if branch on line 98 would have already run. Changing this else to an if should fix this issue.